### PR TITLE
Fix typos and remove some trailing whitespace

### PR DESCRIPTION
**Pull request by invidian**
## Description

Found some typos while working on https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2280#discussion_r869768831, which made me to routinely run `codespell` on this codebase to fix some other typos along the way. Turns out the list is pretty long :sweat_smile:  

I suggest to add a similar GH Action to https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2280 to reduce number of introduced typos in the future. Let me know if I should create an issue for it or a PR, or maybe it can be pushed here?

I can also try to break down this PR somehow for more reviewable chunks, not sure if that make sense though.

I'd also review the ignore lists, as maybe some words can actually be removed and fixed, I put there those which I wasn't sure what to do with, but to make `codespell` not detect anything anymore.

Codespell command to run:
```sh
codespell -S .git,./calico/js,./felix/bpf-gpl/include/libbpf,./.go-pkg-cache,./calico/_site,./calico/_sass,./go.sum,./calico/fonts/font-awesome/fontawesome-webfont.svg -L aks,mmaped,wan,merchantibility,unknwon,synching,parms,fo,ba,mke,buil,parm,nin,auto-detetion,inout,benchs,quater,cnat,afe -f -H
```

I could also suggest rules like this for `semgrep` to reduce number of trailing whitespace in files:
```yaml
rules:
  - id: files-must-not-have-trailing-whitespace
    patterns:
      - pattern-regex: '[[:blank:]]$'
    message: Files must not have any trailing whitespace.
    languages: [generic]
    severity: ERROR
  - id: files-must-not-have-trailing-newlines
    patterns:
      - pattern-regex: '\n\n\Z'
    message: Files must not have any trailing newlines.
    languages: [generic]
    severity: ERROR
  - id: all-lines-must-end-with-newline
    patterns:
      - pattern-regex: '\S\z'
    message: All lines must end with newline.
    languages: [generic]
    severity: ERROR
```

For better consistency in styling, I'd also suggest using tool/linter like `gofumpt`. Currently running `gofumpt -w $(find -name *.go | tr \\n ' '; echo)` shows a lot of inconsistency.

## Release Note

```release-note
TBD
```

## Reminder for the reviewer

Make sure that this PR has the correct labels and milestone set.

Every PR needs one `docs-*` label.

- `docs-pr-required`: This change requires a change to the documentation that has not been completed yet.
- `docs-completed`: This change has all necessary documentation completed.
- `docs-not-required`: This change has no user-facing impact and requires no docs.

Every PR needs one `release-note-*` label.

- `release-note-required`: This PR has user-facing changes. Most PRs should have this label.
- `release-note-not-required`: This PR has no user-facing changes.

Other optional labels:

- `cherry-pick-candidate`: This PR should be cherry-picked to an earlier release. For bug fixes only.
- `needs-operator-pr`: This PR is related to install and requires a corresponding change to the operator.

### CODE REVIEW

**Review by tomastigera — APPROVED** [review]
Skimmed the bpf / felix related changes, LGTM thanks

**Comment by caseydavenport** [comment]
Hm, looks like our formatting check doesn't like some change in calicoctl:

```
./calicoctl/commands/crds/crds.go
../lib.Makefile:562: recipe for target 'check-fmt' failed
```

**Comment by caseydavenport** [comment]
Looks like added whitespace here: https://github.com/projectcalico/calico/pull/6067/files#diff-1a47f581347e92a3df11201d569b99082c5657699d8d531ca8be0ed8e73c5ffb

**Comment by invidian** [comment]
Thanks for checking. I was hoping/expecting that CI will find some eventual inconsistencies. Will fix.

**Comment by marvin-tigera** [comment]
Removing "merge-when-ready" label due to new commits

**Comment by invidian** [comment]
Fixed the template which was adding this whitespace.

**Comment by invidian** [comment]
@caseydavenport and what is your take on the semgrep/linter for formatting? Should I at least open some issues about it?

**Comment by caseydavenport** [comment]
@invidian an issue would be good - I need to look into those tools a bit more closely but in general I'm on board with some gentle spellchecking.

File: calicoctl/calicoctl/commands/crds/crds.go [hunk 1]
```diff
@@ -12,7 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package crds
+package crds
```
Comment (caseydavenport) [inline calicoctl/calicoctl/commands/crds/crds.go:15]: @invidian I think you need to re-run the generation, since the extra space is still here

Comment (invidian) [reply calicoctl/calicoctl/commands/crds/crds.go]: Ugh, of course, just looked at the CI logs. I _think_ it should be OK now.

**Comment by invidian** [comment]
Hmm, from CI logs I can't figure out what's wrong and it _seems_ I cannot access the artifacts either for more information.

**Comment by caseydavenport** [comment]
Giving it a rerun - I believe those might just be test flakes.

**Comment by caseydavenport** [comment]
Ok, looks like the test failures are flakes. However, this PR needs a rebase now due to conflicts.

**Comment by marvin-tigera** [comment]
Removing "merge-when-ready" label due to new commits

**Pull request merged**